src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -280,12 +280,17 @@ private function createFirewalls($config, ContainerBuilder $container)
                 @trigger_error(sprintf('Not setting "logout_on_user_change" to true on firewall "%s" is deprecated as of 3.4, it will always be true in 4.0.', $name), E_USER_DEPRECATED);
             }
 
-            $contextListenerDefinition->addMethodCall('setLogoutOnUserChange', array($firewall['logout_on_user_change']));
-
             $configId = 'security.firewall.map.config.'.$name;
 
             list($matcher, $listeners, $exceptionListener) = $this->createFirewall($container, $name, $firewall, $authenticationProviders, $providerIds, $configId);
 
+            foreach ($listeners as $listener) {
+                if (0 === strpos($listenerId = (string) $listener, 'security.context_listener.')) {
+                    $container->getDefinition($listenerId)->addMethodCall('setLogoutOnUserChange', array($firewall['logout_on_user_change']));
```

Comment:
Can be done in `createContextListener` by passing the option value as argument to createFirewall first, or iterate over `$this->contextListeners` after this loop.

Reply:
@chalasr done :wink: Will make merging into higher branches a bit harder though :yum: